Add support for using (v,e) interpolation tables in THM

GiudGiud opened the pull request:
Bunch of missing routines, enough to be using the 1 phase pipe flow model at least
EDIT: and the volume junction now. So it should be nearly all we need for regular use, and we are within striking distance if a new object comes up!

Also added:
- additional checks on input variables (consistent across the entire object now)
- more AD support. The "capping" of input variables does not play well with our use of "const ADReal &" but I think the capping (for robustness) is more important than performance right now, so I made copies of the ADReal inputs
- more unit testing, covering previously uncovered routines
- deprecated 'fp' and added a workaround as THM simulations love having 'fp' in GlobalParams and 'fp = my_name' does not work for TFP
- widely buffed the ability to do partial passthrough, e.g. interpolating some quantities and using an 'input_fp' for others. It's well tested now, and you can pass-through a tabulated property (previously would not let you)

@licharlot in case you want to keep following TFPs

GiudGiud committed:
Avoid empty columns in tabulation file

GiudGiud committed:
Check the initial guess when loading a (v,e) tabulation
refs #20101

GiudGiud committed:
Make 2D newton inversion able to do 1D solves in degenerate case
Add debugging output

GiudGiud committed:
Add ability to compute rho from p T when using v,e tabulations: using newton
refs #20101

GiudGiud committed:
Better error on missing routines

GiudGiud committed:
Add e_from_p_rho computed from a ve tabulation
With this we can initialze a THM flow channel 1 phase
refs #20101

GiudGiud committed:
Add rho from p T with AD for ve tabulations:
- add the p/T from v e with derivatives and AD derivatives to be able to do a Newton solve with AD
refs #20101

GiudGiud committed:
Add e_from_p_rho with AD for ve tabulations
- requires T_from_p_rho and e_from_p_T with AD for pT tabulations
refs #20101

GiudGiud committed:
Adjust newton2D parameters
- use parameter tolerance
- use parameter number of iterations

GiudGiud committed:
Add the caller name for the 2D newton solves to help debugging from the input file
- also the current iterate and targets

GiudGiud committed:
Add test for degenerate Newton solve

GiudGiud committed:
Add test for new fluid properties calls

GiudGiud committed:
Add more log to all Newton utils : use parameters in SPFP too

GiudGiud committed:
Simplify default params for newton inversion

GiudGiud committed:
Add h_from_p_T using a (v,e) tabulation for volume junctions
refs #20101

GiudGiud committed:
Rework again error messages:
- mention two options: interpolated properties with specific name
                       fp object

GiudGiud committed:
Add another error for missing input

GiudGiud committed:
Add e_from_p_T using a (v,e) tabulation
to enable using volume junctions
- use newton on getting rho=1/v, then newton again to get e by inverting T(v,e)

GiudGiud committed:
Add missing s_from_v_e with derivatives
to use TFP with volume junction
When using (p,T) previously, s_from_v_e ended up computed through a p_T_from_v_e call

GiudGiud committed:
Implement h_from_p_T with derivatives using (v,e) tabulation
to satisfy the p_from_h_s call in volume junction

GiudGiud committed:
Add rho from p T with derivatives
to enable use of v e tabuluations with volume junctions

GiudGiud committed:
Add s from p T with derivatives with v e tabulation

GiudGiud committed:
Use the tabulations in presence of a FP source uo when computing properties (consistency)
Use the FP uo preferentially for computing grids

GiudGiud committed:
Add unit tests for properties from ve computed from a tabulation generated from a FP object

GiudGiud committed:
Label the unit tests better, highlighting the mssing fromVEFromFile case

GiudGiud committed:
Add some missing tests for p,T tabulation and p,T->(v,e) tabulation

GiudGiud committed:
More clarifying what fluid properties object is built how in unit tests

GiudGiud commented:
Worked on adding support for a volume junction, so this should be enough for general use in THM now. 
Bunch of (p, s), (p, h) etc calls work now

GiudGiud committed:
Simplify logic for having ve interpolations
Add more range check

GiudGiud committed:
Deprecate 'fp' parameter to be able to use 'fp' in GlobalParams

GiudGiud committed:
Add a test for (v,e) output of tabulation
Add a  dedicated test for (p,T) output

GiudGiud committed:
Fix FP output from (v,e):
- enable when not using (p,T)->(v,e) conversion
- avoid the header to be able to use for gold files
- avoid a lone comma when not outputting any extra properties
- avoid conversion errors with h(p,T) call

GiudGiud committed:
Allow fp in GlobalParams with TFP

GiudGiud committed:
std::cout -> Moose::out

moosebuild commented:
Job [Test](https://civet.inl.gov/job/3377901/), step Results summary on fd862a6 wanted to post the following:

## Framework test summary

Compared against 7edd10c in job [civet.inl.gov/job/3374418](https://civet.inl.gov/job/3374418).

### No added tests


### Run time changes

| Test                                                                                               | Base (s)   | Head (s)   | +/-     |
|----------------------------------------------------------------------------------------------------|------------|------------|---------|
| `geomsearch/patch_update_strategy.nonlinear_iter`                                                  | 2.98       | 5.62       | +88.85% |
| `meshgenerators/distributed_rectilinear/dmg_displaced_mesh.pbc/adaptivity_autoscaling`             | 3.35       | 6.06       | +81.16% |
| `kernels/ad_mat_diffusion.1D/ad_1D_transient_jacobian`                                             | 2.72       | 4.91       | +80.27% |
| `bcs/ad_penalty_dirichlet_bc.ad/function_penalty_dirichlet_bc_test-jac`                            | 2.36       | 4.21       | +78.58% |
| `outputs/debug.show_execution_order/dirac`                                                         | 2.57       | 4.57       | +78.07% |
| `fvkernels/dispersion-test.top_right_limited_scalar_advection/venkatakrishnan`                     | 3.77       | 6.69       | +77.46% |
| `kernels/vector_fe.coupled_electrostatics_errors/tet14`                                            | 2.48       | 4.38       | +76.79% |
| `geomsearch/patch_update_strategy.always`                                                          | 2.79       | 4.92       | +76.19% |
| `time_integrators/convergence.implicit_astabledirk4/level0`                                        | 2.42       | 4.26       | +76.10% |
| `time_integrators/convergence.implicit_midpoint/level0`                                            | 2.02       | 3.55       | +75.68% |
| `kernels/array_coupled_time_derivative.jac_3d`                                                     | 2.28       | 3.99       | +75.37% |
| `outputs/debug.show_execution_order/regular_kernels`                                               | 2.76       | 4.83       | +74.78% |
| `multiapps/relaxation.sub_relaxed`                                                                 | 2.44       | 4.25       | +74.43% |
| `mesh/named_entities.test_periodic_names`                                                          | 2.05       | 3.57       | +73.93% |
| `geomsearch/3d_moving_penetration.pl_test3`                                                        | 2.10       | 3.64       | +73.83% |
| `outputs/console/multiapp.ts_end_multi_output`                                                     | 3.06       | 5.31       | +73.68% |
| `functormaterials/functor_change.test/nonlinear`                                                   | 2.31       | 4.01       | +73.36% |
| `kernels/ad_mat_diffusion.1D/ad_1D_transient`                                                      | 2.12       | 3.67       | +73.18% |
| `time_integrators/convergence.implicit_lstabledirk3/level0`                                        | 2.07       | 3.59       | +73.17% |
| `executioners/aux-ss-detection.aux-ss`                                                             | 2.09       | 3.62       | +73.13% |
| `multiapps/secant_postprocessor.pp_transient/app_begin_transfers_end_secant_sub`                   | 6.22       | 10.76      | +73.09% |
| `multiapps/steffensen_postprocessor.pp_transient/app_end_transfers_end_steffensen_sub`             | 2.88       | 4.99       | +72.98% |
| `executioners/transient_sync_time.testsynctimes`                                                   | 4.12       | 7.12       | +72.84% |
| `multiapps/picard.iteration_adaptive_picard_max_its`                                               | 2.22       | 3.83       | +72.52% |
| `multiapps/steffensen_postprocessor.pp_transient/app_end_transfers_end`                            | 2.11       | 3.64       | +72.42% |
| `kernels/2d_diffusion.matdiffusion/ad`                                                             | 3.70       | 6.37       | +72.27% |
| `multiapps/multilevel.console_to_file`                                                             | 2.85       | 4.91       | +72.23% |
| `multiapps/steffensen.variables_transient/app_begin_transfers_end_steffensen_sub`                  | 4.66       | 8.03       | +72.22% |
| `time_integrators/convergence.implicit_astabledirk4/level1`                                        | 3.77       | 6.48       | +72.04% |
| `outputs/format.json_search`                                                                       | 13.26      | 22.79      | +71.87% |
| `outputs/debug.show_execution_order/dgkernels`                                                     | 2.54       | 4.36       | +71.78% |
| `materials/derivative_material_interface.ad_analytic_derivatives/material_chaining`                | 21.22      | 36.42      | +71.59% |
| `executioners/nl_divergence_tolerance.test`                                                        | 2.10       | 3.60       | +71.45% |
| `materials/derivative_material_interface.ad_zero_derivatives`                                      | 2.18       | 3.73       | +71.28% |
| `multiapps/secant_postprocessor.pp_transient/app_begin_transfers_begin_secant_sub`                 | 3.18       | 5.44       | +71.14% |
| `time_integrators/dirk.dirk3_fdp`                                                                  | 2.14       | 3.65       | +70.99% |
| `time_integrators/multiple-integrators.jac`                                                        | 3.72       | 6.34       | +70.27% |
| `dirackernels/point_caching.point_caching/moving_mesh`                                             | 2.09       | 3.54       | +69.81% |
| `executioners/multisys-ss-detection.multi-sys-ss`                                                  | 8.58       | 14.57      | +69.67% |
| `time_integrators/convergence.implicit_lstabledirk3/level2`                                        | 5.68       | 9.63       | +69.56% |
| `constraints/overwrite_variables.test_balance`                                                     | 2.76       | 4.68       | +69.56% |
| `mfem/kernels.MFEMIrrotational`                                                                    | 8.68       | 14.71      | +69.43% |
| `time_steppers/fixed_point_iteration_adaptive_dt.increase_dt`                                      | 2.42       | 4.09       | +68.74% |
| `multiapps/steffensen_postprocessor.pp_transient/app_begin_transfers_end_steffensen_sub`           | 4.83       | 8.13       | +68.45% |
| `multiapps/steffensen.variables_transient/app_end_transfers_end_steffensen_sub`                    | 2.87       | 4.84       | +68.30% |
| `multiapps/multilevel.group/time_dt_from_parent`                                                   | 2.82       | 4.75       | +68.20% |
| `multiapps/steffensen.variables_transient/app_begin_transfers_end`                                 | 5.08       | 8.55       | +68.17% |
| `multiapps/secant_postprocessor.pp_transient/app_end_transfers_end`                                | 3.00       | 5.05       | +68.15% |
| `hdgkernels/ip-advection-diffusion.mms/lagrange_quad`                                              | 10.28      | 17.28      | +68.12% |
| `misc/signal_handler.test_signal_recover`                                                          | 2.67       | 4.48       | +67.84% |
| `bcs/ad_function_neumann_bc.ad/test-jac`                                                           | 4.76       | 7.99       | +67.81% |
| `multiapps/secant.variables_transient/app_begin_transfers_end_secant_sub`                          | 7.01       | 11.74      | +67.59% |
| `multiapps/secant_postprocessor.pp_transient/app_begin_transfers_begin`                            | 3.14       | 5.26       | +67.56% |
| `time_integrators/convergence.implicit_astabledirk4/level2`                                        | 6.13       | 10.27      | +67.51% |
| `transfers/multiapp_conservative_transfer.userobject_transfer_csv`                                 | 2.28       | 3.81       | +67.49% |
| `misc/signal_handler.test_signal_recover_exodiff`                                                  | 2.75       | 4.60       | +67.38% |
| `multiapps/picard.rel_tol`                                                                         | 5.89       | 9.83       | +66.93% |
| `kernels/2d_diffusion.matdiffusion_coupled/ad`                                                     | 6.48       | 10.82      | +66.85% |
| `multiapps/picard_postprocessor.pp_transient/app_begin_transfers_begin`                            | 5.81       | 9.69       | +66.85% |
| `multiapps/picard_postprocessor.pp_transient/app_end_transfers_end_picard_sub`                     | 2.91       | 4.86       | +66.65% |
| `convergence/reference_residual_convergence.zero_tolerance_ref`                                    | 8.27       | 13.79      | +66.63% |
| `materials/derivative_sum_material.random_ic/ad_validating`                                        | 24.08      | 40.12      | +66.60% |
| `transfers/multiapp_userobject_transfer.transfer/to_sub_displaced`                                 | 2.25       | 3.74       | +66.55% |
| `materials/derivative_material_interface.ad_analytic_derivatives/derivative_jacobian`              | 6.65       | 11.07      | +66.39% |
| `transfers/multiapp_postprocessor_interpolation_transfer.basis/from_sub_to_parent`                 | 3.18       | 5.30       | +66.35% |
| `outputs/format.json_no_test_objects`                                                              | 14.74      | 24.52      | +66.33% |
| `outputs/debug.show_execution_order/interface_kernels`                                             | 2.69       | 4.47       | +66.22% |
| `multiapps/secant.variables_transient/app_end_transfers_end`                                       | 2.99       | 4.97       | +66.21% |
| `utils/libtorch_nn/ann.libtorch-nn-optimizer/rmsprop`                                              | 2.96       | 4.91       | +65.87% |
| `multiapps/steffensen.variables_transient/test_rerun`                                              | 3.15       | 5.22       | +65.44% |
| `time_integrators/convergence.implicit_lstabledirk3/level1`                                        | 3.51       | 5.81       | +65.39% |
| `kernels/body_force.ad_mat_forcing_function_test`                                                  | 7.67       | 12.68      | +65.34% |
| `geomsearch/3d_moving_penetration_smoothing.overlapping/pl_test4ns`                                | 2.18       | 3.61       | +65.30% |
| `multiapps/secant.variables_transient/test_rerun`                                                  | 3.76       | 6.21       | +65.21% |
| `fviks/diffusion.diffusion-multisystem`                                                            | 3.10       | 5.12       | +65.14% |
| `outputs/debug.show_execution_order/regular_bcs`                                                   | 2.55       | 4.21       | +65.08% |
| `bcs/periodic.auto_wrap_3d_test`                                                                   | 2.37       | 3.92       | +65.07% |
| `outputs/console/multiapp.ts_both_multi_output`                                                    | 4.16       | 6.86       | +65.06% |
| `multiapps/transient_multiapp.group/dt_from_parent`                                                | 2.19       | 3.62       | +65.01% |
| `multiapps/steffensen_postprocessor.pp_transient/iteration_adaptive`                               | 2.44       | 4.02       | +65.00% |
| `parser/cli_multiapp_single.cli_override_single`                                                   | 2.14       | 3.52       | +64.98% |
| `meshmodifiers/element_subdomain_modifier.boundary/refined_boundary_refined_different_levels`      | 2.53       | 4.17       | +64.82% |
| `mfem/submeshes.MFEMGeometryCutTransitionSubMesh`                                                  | 21.11      | 34.78      | +64.77% |
| `meshgenerators/distributed_rectilinear/dmg_displaced_mesh.pbc/adaptivity_nemesis`                 | 2.30       | 3.79       | +64.70% |
| `multiapps/transient_multiapp.test_negative_time`                                                  | 2.22       | 3.66       | +64.66% |
| `outputs/sampled_output.group/test_first_order`                                                    | 4.40       | 7.24       | +64.60% |
| `multiapps/steffensen.variables_transient/app_begin_transfers_begin_steffensen_sub`                | 3.02       | 4.97       | +64.55% |
| `reporters/parsed_reporters.vectorMath`                                                            | 2.60       | 4.27       | +64.47% |
| `geomsearch/3d_moving_penetration_smoothing.disjoint/pl_test3ns`                                   | 2.22       | 3.65       | +64.43% |
| `multiapps/picard_postprocessor.pp_transient/app_end_transfers_end`                                | 5.94       | 9.77       | +64.38% |
| `geomsearch/3d_moving_penetration_smoothing.overlapping/pl_test4qnstt`                             | 2.02       | 3.31       | +64.32% |
| `time_integrators/scalar.group/scalar`                                                             | 8.73       | 14.33      | +64.26% |
| `multiapps/secant_postprocessor.pp_transient/sub_cycling`                                          | 2.94       | 4.83       | +64.08% |
| `geomsearch/3d_moving_penetration.pl_test3q`                                                       | 2.05       | 3.37       | +64.06% |
| `parser/cli_multiapp_all.cli_override_all`                                                         | 3.29       | 5.39       | +64.04% |
| `multiapps/picard_postprocessor.pp_transient/iteration_adaptive`                                   | 5.69       | 9.33       | +63.99% |
| `kokkos/auxkernels/time_integration.time_integration_aux`                                          | 2.21       | 3.62       | +63.86% |
| `fvkernels/mms/advective-outflow.CentralDifferenceLimiter`                                         | 7.90       | 12.94      | +63.83% |
| `misc/signal_handler.test_signal_parallel_recover`                                                 | 2.60       | 4.26       | +63.74% |
| `outputs/console/multiapp.picard_norm_output`                                                      | 3.31       | 5.41       | +63.55% |
| `mfem/submeshes.MFEMDomainSubMeshBoundary`                                                         | 17.51      | 28.64      | +63.53% |
| `problems/reference_residual_problem.zero_tolerance_ref`                                           | 8.62       | 14.09      | +63.44% |
| `transfers/multiapp_postprocessor_interpolation_transfer.basis/from_sub_to_parent_single_sub_file` | 3.34       | 5.46       | +63.44% |
| `functormaterials/functor_change.test/fixed_point`                                                 | 3.94       | 6.44       | +63.42% |
| `meshgenerators/cut_mesh_by_level_set_generator.test/with_xyzd`                                    | 3.81       | 6.22       | +63.38% |
| `hdgkernels/ldg-diffusion.rt_mms/tri`                                                              | 7.74       | 12.64      | +63.30% |
| `dirackernels/front_tracking.test`                                                                 | 2.74       | 4.46       | +63.17% |
| `fvkernels/block-restriction.jac`                                                                  | 2.28       | 3.72       | +63.15% |
| `transfers/multiapp_nearest_node_transfer/cached_multiple_apps.caching_to_multiple_subapps`        | 2.23       | 3.64       | +63.09% |
| `materials/derivative_sum_material.random_ic/validating`                                           | 5.07       | 8.26       | +63.07% |
| `outputs/format.json_no_template`                                                                  | 15.44      | 25.14      | +62.83% |
| `transfers/multiapp_variable_value_sample_transfer.from_multiapp_set_primary_var_error`            | 2.25       | 3.66       | +62.82% |
| `kernels/2d_diffusion.matdiffusion_coupled/nonad`                                                  | 2.02       | 3.28       | +62.79% |
| `parser/cli_multiapp_group.cli_override_group`                                                     | 3.26       | 5.30       | +62.75% |
| `materials/derivative_material_interface.postprocessor_coupling/ad_parsed_material`                | 2.32       | 3.77       | +62.69% |
| `outputs/format.json_line_info`                                                                    | 16.51      | 26.85      | +62.57% |
| `postprocessors/discrete_variable_residual_norm.test`                                              | 5.73       | 9.31       | +62.56% |
| `materials/derivative_material_interface.additional_derivatives/mat_props`                         | 3.49       | 5.68       | +62.50% |
| `materials/compile_time_derivative.test`                                                           | 6.80       | 11.04      | +62.41% |
| `mesh/uniform_refine.square`                                                                       | 3.10       | 5.04       | +62.33% |
| `postprocessors/mms_polynomial.test`                                                               | 2.49       | 4.04       | +62.19% |
| `fvkernels/dispersion-test.bottom_left_limited_scalar_advection/venkatakrishnan`                   | 4.10       | 6.65       | +62.08% |
| `multiapps/steffensen.variables_transient/sub_cycling`                                             | 2.83       | 4.59       | +62.02% |
| `neml2/torchscript.heat_conduction`                                                                | 2.03       | 3.29       | +61.98% |
| `bcs/periodic.auto_wrap_2d_test`                                                                   | 2.34       | 3.78       | +61.86% |
| `outputs/format.definition_scraping_markers`                                                       | 16.04      | 25.95      | +61.76% |
| `functors/scalar_variable.test`                                                                    | 2.30       | 3.72       | +61.68% |
| `multiapps/steffensen_postprocessor.pp_transient/app_begin_transfers_begin`                        | 2.47       | 3.99       | +61.48% |
| `outputs/format.json_full`                                                                         | 15.52      | 25.05      | +61.41% |
| `multiapps/picard_postprocessor.pp_transient/test_rerun`                                           | 4.91       | 7.92       | +61.38% |
| `multiapps/quadrature_point_multiapp.test`                                                         | 2.46       | 3.98       | +61.35% |
| `multiapps/picard_postprocessor.pp_transient/app_begin_transfers_begin_picard_sub`                 | 3.18       | 5.13       | +61.31% |
| `multiapps/positions_from_file.dt_from_multi`                                                      | 2.11       | 3.41       | +61.29% |
| `fviks/one-var-diffusion.first_order_convergence/with-ik`                                          | 5.81       | 9.38       | +61.28% |
| `geomsearch/3d_moving_penetration.pl_test3tt`                                                      | 2.19       | 3.53       | +61.17% |
| `multiapps/secant.variables_transient/iteration_adaptive`                                          | 3.39       | 5.47       | +61.14% |
| `misc/line_source.test`                                                                            | 2.49       | 4.02       | +61.07% |
| `fvkernels/mms/advective-outflow.kt/KTLimitedUpwind`                                               | 15.60      | 25.12      | +61.06% |
| `linearfvbcs/robin.advection-2d-robin`                                                             | 5.50       | 8.86       | +60.93% |
| `postprocessors/side_integral.functors/error_no_face_info`                                         | 2.38       | 3.82       | +60.92% |
| `materials/derivative_material_interface.ad_analytic_derivatives/jacobian`                         | 2.34       | 3.77       | +60.88% |
| `hdgkernels/ip-advection-diffusion.mms/lagrange_tri`                                               | 7.62       | 12.25      | +60.83% |
| `kernels/functor_kernel.test_add`                                                                  | 3.89       | 6.25       | +60.76% |
| `multiapps/secant.variables_transient/app_begin_transfers_begin_secant_sub`                        | 3.76       | 6.04       | +60.73% |
| `geomsearch/3d_moving_penetration_smoothing.disjoint/pl_test3nstt`                                 | 2.25       | 3.61       | +60.71% |
| `multiapps/picard.iteration_adaptive`                                                              | 6.53       | 10.50      | +60.67% |
| `transfers/transfer_interpolation.test`                                                            | 3.71       | 5.95       | +60.54% |
| `multiapps/steffensen.variables_transient/iteration_adaptive`                                      | 3.36       | 5.39       | +60.53% |
| `interfacekernels/1d_interface.mixed_shapes_jacobian_test`                                         | 2.49       | 3.99       | +60.51% |
| `outputs/format.definition_default_type_test`                                                      | 16.11      | 25.86      | +60.47% |
| `bcs/nodal_normals.cylinder_hexes`                                                                 | 2.31       | 3.71       | +60.47% |
| `convergence/postprocessor_convergence.multiapp_fp`                                                | 4.33       | 6.95       | +60.38% |
| `transfers/general_field/user_object/duplicated_user_object_tests.transfer/to_sub`                 | 2.21       | 3.55       | +60.37% |
| `postprocessors/mms_slope.test`                                                                    | 2.13       | 3.42       | +60.27% |
| `interfacekernels/2d_interface.jacobian_test`                                                      | 2.57       | 4.12       | +60.24% |
| `transfers/coord_transform/both-transformed/pp_interpolation.both-transformed`                     | 8.02       | 12.84      | +60.12% |
| `linearfvkernels/block-restriction.mms-adr`                                                        | 6.32       | 10.11      | +59.98% |
| `misc/signal_handler.test_signal_parallel_recover_exodiff`                                         | 2.92       | 4.68       | +59.94% |
| `outputs/format.definition_type_sub_test`                                                          | 16.54      | 26.44      | +59.83% |
| `bcs/periodic.auto_wrap_2d_test_non_generated`                                                     | 2.55       | 4.08       | +59.79% |
| `fvkernels/mms/advective-outflow.MinModLimiter`                                                    | 9.88       | 15.78      | +59.78% |
| `multiapps/secant.variables_transient/app_end_transfers_end_secant_sub`                            | 2.80       | 4.48       | +59.75% |
| `outputs/nemesis.nemesis_scalar_replicated`                                                        | 2.21       | 3.52       | +59.72% |
| `misc/static_condensation.mms`                                                                     | 5.57       | 8.89       | +59.63% |
| `linearfvkernels/diffusion-reaction-advection.mms-adr-2d-dirichlet`                                | 7.10       | 11.34      | +59.62% |
| `preconditioners/fdp.jacobian_fdp_coloring_diagonal_test_fail`                                     | 2.59       | 4.14       | +59.49% |
| `multiapps/steffensen.variables_transient/app_end_transfers_end`                                   | 2.74       | 4.37       | +59.47% |
| `bcs/mat_neumann_bc.ad_test`                                                                       | 2.78       | 4.42       | +59.38% |
| `hdgkernels/ldg-diffusion.mms/quad`                                                                | 12.89      | 20.54      | +59.35% |
| `kernels/vector_fe.coupled_electrostatics/tet14`                                                   | 3.03       | 4.84       | +59.34% |
| `fvkernels/mms/skewness-correction/adv-diff-react.skewcorrected-advection`                         | 6.65       | 10.60      | +59.33% |
| `postprocessors/nodal_extreme_value.nodal_max/test_nodal_max`                                      | 2.02       | 3.21       | +59.20% |
| `multiapps/steffensen_postprocessor.pp_transient/app_begin_transfers_begin_steffensen_sub`         | 2.26       | 3.60       | +59.20% |
| `fvkernels/mms/advective-outflow.VanLeerLimiter`                                                   | 9.83       | 15.64      | +59.18% |
| `hdgkernels/ldg-diffusion.rt_mms/quad`                                                             | 12.16      | 19.32      | +58.93% |
| `adaptivity/displaced_mesh.linear_lagrange_source`                                                 | 2.06       | 3.27       | +58.74% |
| `geomsearch/3d_moving_penetration_smoothing.disjoint/pl_test3nns`                                  | 2.29       | 3.63       | +58.71% |
| `multiapps/picard_multilevel.test_force_norms`                                                     | 2.41       | 3.82       | +58.66% |
| `meshgenerators/distributed_rectilinear/generator.3D_ptscotch`                                     | 2.07       | 3.29       | +58.57% |
| `multiapps/steffensen.variables_transient/app_begin_transfers_begin`                               | 3.15       | 5.00       | +58.50% |
| `meshgenerators/distributed_rectilinear/dmg_displaced_mesh.pbc_adaptivity`                         | 3.41       | 5.40       | +58.48% |
| `multiapps/picard_postprocessor.pp_transient/app_begin_transfers_end_picard_sub`                   | 5.00       | 7.92       | +58.37% |
| `multiapps/relaxation.parent_relaxed`                                                              | 2.38       | 3.76       | +58.32% |
| `misc/ad_robustness.neighbor_ad_with_aux_jac`                                                      | 2.96       | 4.69       | +58.26% |
| `ics/function_scalar_ic.test`                                                                      | 2.06       | 3.27       | +58.25% |
| `geomsearch/3d_moving_penetration_smoothing.disjoint/pl_test3nnstt`                                | 2.25       | 3.55       | +58.23% |
| `bcs/periodic.testperiodic_dp`                                                                     | 3.55       | 5.61       | +58.23% |
| `multiapps/cliargs_from_file.multiple_parameters`                                                  | 2.20       | 3.48       | +58.11% |
| `kernels/ad_flux_divergence.transient`                                                             | 2.14       | 3.39       | +57.96% |
| `fvkernels/mms/advective-outflow.kt/KTLimitedCD`                                                   | 12.28      | 19.38      | +57.88% |
| `multiapps/secant.variables_transient/sub_cycling`                                                 | 3.09       | 4.87       | +57.87% |
| `positions/creating_multiapps.error/size_changed`                                                  | 2.36       | 3.72       | +57.85% |
| `materials/derivative_material_interface.analytic_derivatives/material_chaining`                   | 5.30       | 8.37       | +57.83% |
| `transfers/multiapp_userobject_transfer.transfer/from_sub_nearest_sub_app`                         | 2.28       | 3.60       | +57.76% |
| `linearfvkernels/advection.mms-2d-linear`                                                          | 7.24       | 11.42      | +57.65% |
| `time_integrators/convergence.implicit_lstabledirk4/level2`                                        | 2.04       | 3.21       | +57.49% |
| `linearfvkernels/diffusion-reaction-advection.mms-adr-2d-outflow`                                  | 7.18       | 11.29      | +57.40% |
| `time_integrators/dirk.dirk4_fdp`                                                                  | 3.24       | 5.09       | +57.38% |
| `functormaterials/output.types/ad`                                                                 | 2.19       | 3.45       | +57.37% |
| `fvkernels/mms/advective-outflow.kt/KTLimitedVanLeer`                                              | 12.35      | 19.42      | +57.30% |
| `multiapps/cliargs_from_file.multiline_file`                                                       | 2.12       | 3.34       | +57.30% |
| `outputs/console/multiapp.ts_begin_multi_output`                                                   | 3.46       | 5.43       | +57.21% |
| `time_integrators/convergence.implicit_midpoint/level1`                                            | 3.03       | 4.76       | +57.16% |
| `outputs/csv.all_columns`                                                                          | 4.01       | 6.31       | +57.16% |
| `transfers/general_field/user_object/duplicated_user_object_tests.transfer/to_sub_displaced`       | 2.29       | 3.59       | +57.00% |
| `kernels/scaled_coupled_time_derivative.ad_scaled_coupled_time_derivative_test`                    | 2.31       | 3.62       | +56.84% |
| `multiapps/picard_postprocessor.pp_transient/sub_cycling`                                          | 5.37       | 8.41       | +56.72% |
| `geomsearch/patch_update_strategy.auto`                                                            | 2.65       | 4.15       | +56.43% |
| `meshgenerators/distributed_rectilinear/generator.3D`                                              | 2.18       | 3.40       | +56.40% |
| `multiapps/picard.abs_tol`                                                                         | 5.77       | 9.02       | +56.33% |
| `multiapps/picard_postprocessor.pp_transient/app_begin_transfers_end`                              | 6.89       | 10.77      | +56.29% |
| `multiapps/secant_postprocessor.pp_transient/test_rerun`                                           | 3.34       | 5.21       | +56.11% |
| `utils/libtorch_nn/ann.libtorch-nn-optimizer/sgd`                                                  | 2.79       | 4.35       | +56.11% |
| `linearfvbcs/robin.advection-2d-robin-nonorthogonal`                                               | 5.83       | 9.09       | +55.96% |
| `dgkernels/passive-scalar-channel-flow.test_jacbian`                                               | 2.13       | 3.32       | +55.82% |
| `geomsearch/patch_update_strategy.never_warning`                                                   | 3.33       | 5.18       | +55.78% |
| `mfem/lor.MFEMHeatConductionLOR`                                                                   | 7.07       | 11.00      | +55.61% |
| `outputs/format.hit`                                                                               | 20.38      | 31.67      | +55.44% |
| `outputs/format.definition_boolean_type_valenum_choices_test`                                      | 16.81      | 26.13      | +55.42% |
| `time_integrators/bdf2.group/test`                                                                 | 3.01       | 4.68       | +55.38% |
| `hdgkernels/ip-advection-diffusion.mms/hierarchic_quad`                                            | 10.26      | 15.94      | +55.35% |
| `kernels/ode.test_parsed_sys_impl`                                                                 | 3.05       | 4.74       | +55.33% |
| `transfers/general_field/user_object/duplicated_user_object_tests.transfer/from_sub`               | 2.26       | 3.50       | +55.07% |
| `kernels/ad_flux_divergence.nonlinear_diffusivity`                                                 | 2.48       | 3.85       | +55.05% |
| `multiapps/steffensen_postprocessor.pp_transient/sub_cycling`                                      | 2.39       | 3.70       | +55.03% |
| `time_integrators/convergence.implicit_midpoint/level2`                                            | 4.86       | 7.52       | +54.76% |
| `multiapps/secant.variables_transient/app_begin_transfers_begin`                                   | 3.51       | 5.43       | +54.67% |
| `controls/time_periods/constraints.test`                                                           | 2.21       | 3.42       | +54.61% |
| `fvkernels/mms/advective-outflow.outflow/average`                                                  | 29.46      | 45.48      | +54.38% |
| `convergence/default_steady_state_convergence.default`                                             | 2.65       | 4.09       | +54.36% |
| `fvkernels/mms/advective-outflow.SOULimiter`                                                       | 10.06      | 15.53      | +54.32% |
| `fvkernels/mms/advective-outflow.extrapolate`                                                      | 8.14       | 12.56      | +54.30% |
| `geomsearch/3d_moving_penetration_smoothing.disjoint/pl_test3qns`                                  | 2.49       | 3.84       | +54.27% |
| `utils/libtorch_nn/ann.libtorch-nn-optimizer/adam`                                                 | 3.10       | 4.78       | +54.16% |
| `fviks/one-var-diffusion.first_order_convergence/without-ik`                                       | 5.59       | 8.61       | +54.15% |
| `linearfvkernels/diffusion.mms-diffusion-2d-nonorthogonal`                                         | 6.51       | 10.03      | +54.01% |
| `outputs/format.definition_active_parameter_test`                                                  | 17.10      | 26.32      | +53.94% |
| `make_install.test_copy_install/run`                                                               | 4.70       | 7.24       | +53.94% |
| `multiapps/secant_postprocessor.pp_transient/iteration_adaptive`                                   | 3.43       | 5.29       | +53.90% |
| `postprocessors/element_variable_value.elem_var_value_fv_test`                                     | 2.38       | 3.66       | +53.78% |
| `mortar/convergence-studies/fv-gap-conductance.fv/second_order`                                    | 9.32       | 14.33      | +53.78% |
| `fviks/one-var-diffusion.harmonic`                                                                 | 6.02       | 9.26       | +53.76% |
| `hdgkernels/ip-advection-diffusion.mms/hierarchic_tri`                                             | 7.07       | 10.86      | +53.68% |
| `multiapps/relaxation.standard`                                                                    | 2.62       | 4.03       | +53.68% |
| `outputs/format.definition_valenum_test`                                                           | 17.00      | 26.12      | +53.60% |
| `outputs/format.no_double_executioner_output`                                                      | 13.58      | 20.85      | +53.52% |
| `outputs/format.hit_search`                                                                        | 14.17      | 21.73      | +53.33% |
| `hdgkernels/ldg-diffusion.mms/tri`                                                                 | 7.99       | 12.25      | +53.29% |
| `fvkernels/mms/grad-reconstruction.mat-cartesian`                                                  | 2.12       | 3.25       | +53.12% |
| `mortar/convergence-studies/fv-gap-conductance.fv/first_order`                                     | 9.29       | 14.21      | +52.99% |
| `preconditioners/fdp.jacobian_fdp_coloring_full_test`                                              | 2.39       | 3.65       | +52.92% |
| `fvkernels/mms/advective-outflow.UpwindLimiter`                                                    | 8.25       | 12.61      | +52.76% |
| `preconditioners/fdp.jacobian_fdp_standard_test`                                                   | 2.57       | 3.92       | +52.69% |
| `fvkernels/mms/skewness-correction/adv-diff-react.skewcorrected`                                   | 6.87       | 10.49      | +52.57% |
| `kernels/body_force.mat_forcing_function_test`                                                     | 2.12       | 3.23       | +52.56% |
| `hdgkernels/ip-advection-diffusion.mms/monomial_quad`                                              | 10.68      | 16.30      | +52.55% |
| `linearfvkernels/advection.mms-2d-upwind`                                                          | 7.71       | 11.75      | +52.48% |
| `linearfvkernels/block-restriction.mms-diffusion`                                                  | 6.09       | 9.27       | +52.37% |
| `misc/jacobian.simple`                                                                             | 4.73       | 7.21       | +52.36% |
| `linearfvkernels/diffusion.mms-diffusion-2d-nonorthogonal_neumann`                                 | 6.96       | 10.58      | +52.07% |
| `mfem/lor.MFEMGradDivLOR`                                                                          | 6.15       | 9.36       | +52.06% |
| `fvkernels/mms/harmonic_interpolation.average/tris`                                                | 6.83       | 10.38      | +52.00% |
| `outputs/format.definition_normal_sub_test`                                                        | 17.11      | 26.01      | +51.98% |
| `linearfvkernels/diffusion.mms-diffusion-2d-rz`                                                    | 6.44       | 9.78       | +51.85% |
| `executioners/eigen_executioners.test_inverse_power_method`                                        | 3.97       | 6.04       | +51.83% |
| `batch_material.test`                                                                              | 2.71       | 4.11       | +51.76% |
| `outputs/format.definition_mesh_file_parameter_requirement_removal_test`                           | 17.10      | 25.94      | +51.70% |
| `multiapps/secant.variables_transient/app_begin_transfers_end`                                     | 7.56       | 11.46      | +51.61% |
| `bcs/dmg_periodic.2d`                                                                              | 2.38       | 3.60       | +51.56% |
| `outputs/format.definition_default_subblock_types_child_parameter_promotion_test`                  | 16.97      | 25.70      | +51.50% |
| `outputs/format.definition_input_choices_test`                                                     | 16.98      | 25.71      | +51.39% |
| `fvkernels/dispersion-test.top_right_limited_scalar_advection/quick`                               | 2.05       | 3.10       | +51.30% |
| `outputs/format.definition_childatleastone_test`                                                   | 17.21      | 26.01      | +51.11% |
| `multiapps/steffensen_postprocessor.pp_transient/app_begin_transfers_end`                          | 5.21       | 7.86       | +51.03% |
| `materials/derivative_material_interface.ordering/ad_construction`                                 | 5.61       | 8.46       | +50.79% |
| `fvkernels/mms/harmonic_interpolation.harmonic/tris`                                               | 6.81       | 10.26      | +50.69% |
| `multiapps/picard.test`                                                                            | 6.07       | 9.14       | +50.64% |
| `samplers/distribute.scale/execute`                                                                | 8.51       | 12.82      | +50.62% |
| `convergence/default_multiapp_fixed_point_convergence.residual`                                    | 4.18       | 6.29       | +50.45% |
| `multiapps/relaxation.array_relaxed`                                                               | 4.04       | 6.07       | +50.42% |
| `transfers/multiapp_userobject_transfer.transfer/to_sub`                                           | 2.49       | 3.74       | +50.39% |
| `meshgenerators/parsed_extra_element_id_generator.parsed_elem_id_generator`                        | 2.06       | 3.10       | +50.36% |
| `kernels/ad_mat_reaction.ad_mat_reaction`                                                          | 2.10       | 3.16       | +50.33% |
| `time_integrators/multiple-integrators.test_accuracy`                                              | 7.68       | 11.54      | +50.27% |
| `bcs/ad_penalty_dirichlet_bc.ad/penalty_dirichlet_bc_test-jac`                                     | 2.61       | 3.92       | +50.09% |
## Modules test summary

ERROR: Results do not exist for event 265741

GiudGiud committed:
Address test failure:
- try to guess when newton is used for initial guess checks

GiudGiud committed:
Fix test failures:
initialize FP earlier & use the tabulation rather than the FP

GiudGiud committed:
Add the ability to ignore a fluid property from the tabulation file by sepecifying the
interpolated properties in the input
- address porous flow failure that was relying on forwarding a property

moosebuild commented:
Job [Test](https://civet.inl.gov/job/3381238/), step Results summary on fac1771 wanted to post the following:

## Framework test summary

Compared against 7edd10c in job [civet.inl.gov/job/3374418](https://civet.inl.gov/job/3374418).

No change
## Modules test summary

Compared against 7edd10c in job [civet.inl.gov/job/3374418](https://civet.inl.gov/job/3374418).

### Added tests

| Test                                              | Time (s)   |
|---------------------------------------------------|------------|
| `fluid_properties/test:tabulated.file_writing/pT` | 1.00       |

### Run time changes

| Test                                          | Base (s)   | Head (s)   | +/-     |
|-----------------------------------------------|------------|------------|---------|
| `porous_flow/test:fluidstate.theis_tabulated` | 7.83       | 3.08       | -60.69% |

GiudGiud committed:
Avoid seg fault on OOB access
Make sure to initialize (v,e) grid for passthrough case
Add testing for (v,e) passthrough

GiudGiud committed:
Add AD (p,T) passthrough testing:
- create a dedicated passthrough tabulation
- use idg fps that have the most AD definitions created

GiudGiud committed:
Use the interpolation bounds from the files in (v,e) over querying the FP
- otherwise we would compute min/max from FP calls OOB from the tabulation
- then flush to min/max that could be OOB

GiudGiud committed:
Examine coverage report and catch most misses:
- T/p from v e with derivatives in the passthrough _fp case
- AD e and v from p T
- e from p rho using a pT tabulation
- AD s from v e
- T from p rho using a ve tabulation

joshuahansel reviewed (changes requested):
On behalf of THM users, thank you!

GiudGiud committed:
Address review

cleaner indexing in newton method 1D degenerate from 2D option

Co-authored-by: joshuahansel <joshua.hansel@inl.gov>

GiudGiud committed:
Update past AD changes for std library functions
